### PR TITLE
Parent node metadata

### PR DESCRIPTION
**Pull request by jimmylai**
## Summary
Sometimes user may want to access the parent `CSTNode`
for more information or traverse in bottom-up manner.
We provide `ParentNodeProvider` for those use cases.
## Test Plan
Unit tests.

### CODE REVIEW

File: libcst/test_scope.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+            enumerate(var)
+
+    def g(self):
+        C.var = 3
+
+
+var = 4
+
+x = C()
+x.var = 5
```
Comment (DragonMinded) [inline libcst/test_scope.py]: What is this file?

File: libcst/metadata/tests/test_parent_node_provider.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                {elt for target in iterator if condition}
+                {elt: target for target in iterator if condition}
+                (elt for target in iterator if condition)
+                """,
+            ),
+        )
+    )
+    def test_parent_node_provier(self, code: str) -> None:
+        wrapper = MetadataWrapper(cst.parse_module(dedent(code)))
+        wrapper.visit(DependentVisitor(test=self))
```
Comment (DragonMinded) [inline libcst/metadata/tests/test_parent_node_provider.py]: Nice trick ;)

**Comment by DragonMinded** [comment]
This seems to combine some unrelated changes. I'm not sure what test_scope.py is but it looks like test code left in on accident. Also, what are the METADATA files and how do they relate? Back to you for comments.

File: libcst/test_scope.py (hunk 1 above)
Comment (jimmylai) [reply libcst/test_scope.py]: This wasn't needed. Just removed it.

File: pip-wheel-metadata/libcst.dist-info/LICENSE [hunk 3]
```diff
@@ -0,0 +1,21 @@
+MIT License
```
Comment (DragonMinded) [inline pip-wheel-metadata/libcst.dist-info/LICENSE]: I'm still confused on this file?

File: pip-wheel-metadata/libcst.dist-info/METADATA [hunk 4]
```diff
@@ -0,0 +1,217 @@
+Metadata-Version: 2.1
```
Comment (DragonMinded) [inline pip-wheel-metadata/libcst.dist-info/METADATA]: Same comment here.

File: pip-wheel-metadata/libcst.dist-info/LICENSE (hunk 3 above)
Comment (jimmylai) [reply pip-wheel-metadata/libcst.dist-info/LICENSE]: I'm sorry, accidentally committed those. Just removed them.

**Pull request merged**